### PR TITLE
fix: sensor: veml7700

### LINKED ISSUE #69114: [Coverity CID: 347209] Untrusted value as argument in drivers/sensor/veml7700/veml7700.c
Static code scan issues found in file:

https://github.com/zephyrproject-rtos/zephyr/tree/adeaa63bb71aba5c1732816df21f40d654e26cd4/drivers/sensor/veml7700/veml7700.c#L557

Category: Insecure data handling
Function: `veml7700_init`
Component: Drivers
CID: [347209](https://scan9.scan.coverity.com/#/project-view/29271/12996?selectedIssue=347209)

Details:

https://github.com/zephyrproject-rtos/zephyr/blob/adeaa63bb71aba5c1732816df21f40d654e26cd4/drivers/sensor/veml7700/veml7700.c#L557

```
551     
552      ret = veml7700_write_thresh_high(dev);
553      if (ret < 0) {
554              return ret;
555      }
556     
>>>     CID 347209:  Insecure data handling  (TAINTED_SCALAR)
>>>     Passing tainted expression "*dev->data" to "veml7700_write_als_conf", which uses it as an offset.
557      ret = veml7700_write_als_conf(dev);
558      if (ret < 0) {
559              return ret;
560      }
561     
562      return 0;
```

For more information about the violation, check the [Coverity Reference](https://scan9.scan.coverity.com/doc/en/cov_checker_ref.html#static_checker_TAINTED_SCALAR). ([CWE-20](http://cwe.mitre.org/data/definitions/20.html))

Please fix or provide comments in coverity using the link:

https://scan9.scan.coverity.com/#/project-view/29271/12996?selectedIssue=347209

Note: This issue was created automatically. Priority was set based on classification
of the file affected and the impact field in coverity. Assignees were set using the MAINTAINERS file.

### LINKED ISSUE #69116: [Coverity CID: 347200] Untrusted value as argument in drivers/sensor/veml7700/veml7700.c
Static code scan issues found in file:

https://github.com/zephyrproject-rtos/zephyr/tree/adeaa63bb71aba5c1732816df21f40d654e26cd4/drivers/sensor/veml7700/veml7700.c

Category: Insecure data handling
Function: `veml7700_attr_set`
Component: Drivers
CID: [347200](https://scan9.scan.coverity.com/#/project-view/29271/12996?selectedIssue=347200)

Details:

https://github.com/zephyrproject-rtos/zephyr/blob/adeaa63bb71aba5c1732816df21f40d654e26cd4/drivers/sensor/veml7700/veml7700.c

Please fix or provide comments in coverity using the link:

https://scan9.scan.coverity.com/#/project-view/29271/12996?selectedIssue=347200

For more information about the violation, check the [Coverity Reference](https://scan9.scan.coverity.com/doc/en/cov_checker_ref.html#static_checker_TAINTED_SCALAR). ([CWE-20](http://cwe.mitre.org/data/definitions/20.html))

Note: This issue was created automatically. Priority was set based on classification
of the file affected and the impact field in coverity. Assignees were set using the MAINTAINERS file.

### LINKED ISSUE #69120: [Coverity CID: 347173] Untrusted value as argument in drivers/sensor/veml7700/veml7700.c
Static code scan issues found in file:

https://github.com/zephyrproject-rtos/zephyr/tree/adeaa63bb71aba5c1732816df21f40d654e26cd4/drivers/sensor/veml7700/veml7700.c

Category: Insecure data handling
Function: `veml7700_write_als_conf`
Component: Drivers
CID: [347173](https://scan9.scan.coverity.com/#/project-view/29271/12996?selectedIssue=347173)

Details:

https://github.com/zephyrproject-rtos/zephyr/blob/adeaa63bb71aba5c1732816df21f40d654e26cd4/drivers/sensor/veml7700/veml7700.c#L233

Please fix or provide comments in coverity using the link:

https://scan9.scan.coverity.com/#/project-view/29271/12996?selectedIssue=347173

For more information about the violation, check the [Coverity Reference](https://scan9.scan.coverity.com/doc/en/cov_checker_ref.html#static_checker_TAINTED_SCALAR). ([CWE-20](http://cwe.mitre.org/data/definitions/20.html))

Note: This issue was created automatically. Priority was set based on classification
of the file affected and the impact field in coverity. Assignees were set using the MAINTAINERS file.

### LINKED ISSUE #69123: [Coverity CID: 347103] Untrusted value as argument in drivers/sensor/veml7700/veml7700.c
Static code scan issues found in file:

https://github.com/zephyrproject-rtos/zephyr/tree/adeaa63bb71aba5c1732816df21f40d654e26cd4/drivers/sensor/veml7700/veml7700.c

Category: Insecure data handling
Function: `veml7700_sample_fetch`
Component: Drivers
CID: [347103](https://scan9.scan.coverity.com/#/project-view/29271/12996?selectedIssue=347103)

Details:

https://github.com/zephyrproject-rtos/zephyr/blob/adeaa63bb71aba5c1732816df21f40d654e26cd4/drivers/sensor/veml7700/veml7700.c

Please fix or provide comments in coverity using the link:

https://scan9.scan.coverity.com/#/project-view/29271/12996?selectedIssue=347103

For more information about the violation, check the [Coverity Reference](https://scan9.scan.coverity.com/doc/en/cov_checker_ref.html#static_checker_TAINTED_SCALAR). ([CWE-20](http://cwe.mitre.org/data/definitions/20.html))

Note: This issue was created automatically. Priority was set based on classification
of the file affected and the impact field in coverity. Assignees were set using the MAINTAINERS file.

### LINKED ISSUE #69124: [Coverity CID: 347099] Untrusted value as argument in drivers/sensor/veml7700/veml7700.c
Static code scan issues found in file:

https://github.com/zephyrproject-rtos/zephyr/tree/adeaa63bb71aba5c1732816df21f40d654e26cd4/drivers/sensor/veml7700/veml7700.c

Category: Insecure data handling
Function: `veml7700_perform_single_measurement`
Component: Drivers
CID: [347099](https://scan9.scan.coverity.com/#/project-view/29271/12996?selectedIssue=347099)

Details:

https://github.com/zephyrproject-rtos/zephyr/blob/adeaa63bb71aba5c1732816df21f40d654e26cd4/drivers/sensor/veml7700/veml7700.c#L411

Please fix or provide comments in coverity using the link:

https://scan9.scan.coverity.com/#/project-view/29271/12996?selectedIssue=347099

For more information about the violation, check the [Coverity Reference](https://scan9.scan.coverity.com/doc/en/cov_checker_ref.html#static_checker_TAINTED_SCALAR). ([CWE-20](http://cwe.mitre.org/data/definitions/20.html))

Note: This issue was created automatically. Priority was set based on classification
of the file affected and the impact field in coverity. Assignees were set using the MAINTAINERS file.

### LINKED ISSUE #69127: [Coverity CID: 347069] Untrusted value as argument in drivers/sensor/veml7700/veml7700.c
Static code scan issues found in file:

https://github.com/zephyrproject-rtos/zephyr/tree/adeaa63bb71aba5c1732816df21f40d654e26cd4/drivers/sensor/veml7700/veml7700.c

Category: Insecure data handling
Function: `veml7700_set_shutdown_flag`
Component: Drivers
CID: [347069](https://scan9.scan.coverity.com/#/project-view/29271/12996?selectedIssue=347069)

Details:

https://github.com/zephyrproject-rtos/zephyr/blob/adeaa63bb71aba5c1732816df21f40d654e26cd4/drivers/sensor/veml7700/veml7700.c#L273

Please fix or provide comments in coverity using the link:

https://scan9.scan.coverity.com/#/project-view/29271/12996?selectedIssue=347069

For more information about the violation, check the [Coverity Reference](https://scan9.scan.coverity.com/doc/en/cov_checker_ref.html#static_checker_TAINTED_SCALAR). ([CWE-20](http://cwe.mitre.org/data/definitions/20.html))

Note: This issue was created automatically. Priority was set based on classification
of the file affected and the impact field in coverity. Assignees were set using the MAINTAINERS file.

### LINKED ISSUE #69128: [Coverity CID: 347000] Untrusted value as argument in drivers/sensor/veml7700/veml7700.c
Static code scan issues found in file:

https://github.com/zephyrproject-rtos/zephyr/tree/adeaa63bb71aba5c1732816df21f40d654e26cd4/drivers/sensor/veml7700/veml7700.c

Category: Insecure data handling
Function: `veml7700_fetch_als`
Component: Drivers
CID: [347000](https://scan9.scan.coverity.com/#/project-view/29271/12996?selectedIssue=347000)

Details:

https://github.com/zephyrproject-rtos/zephyr/blob/adeaa63bb71aba5c1732816df21f40d654e26cd4/drivers/sensor/veml7700/veml7700.c

Please fix or provide comments in coverity using the link:

https://scan9.scan.coverity.com/#/project-view/29271/12996?selectedIssue=347000

For more information about the violation, check the [Coverity Reference](https://scan9.scan.coverity.com/doc/en/cov_checker_ref.html#static_checker_TAINTED_SCALAR). ([CWE-20](http://cwe.mitre.org/data/definitions/20.html))

Note: This issue was created automatically. Priority was set based on classification
of the file affected and the impact field in coverity. Assignees were set using the MAINTAINERS file.

### PR DESCRIPTION
**Pull request by jilaypandya**
This commit introduces checks in order to sanitize input data.
Following Issues should be resolved with this change.

Fixes https://github.com/zephyrproject-rtos/zephyr/issues/69114
Fixes https://github.com/zephyrproject-rtos/zephyr/issues/69116
Fixes https://github.com/zephyrproject-rtos/zephyr/issues/69120
Fixes https://github.com/zephyrproject-rtos/zephyr/issues/69123
Fixes https://github.com/zephyrproject-rtos/zephyr/issues/69124
Fixes https://github.com/zephyrproject-rtos/zephyr/issues/69127
Fixes https://github.com/zephyrproject-rtos/zephyr/issues/69128

### CODE REVIEW

**Comment by jilaypandya** [comment]
Hi @jeffwelder-ellenbytech, could you quickly test out the changes on your hardware?

**Comment by jeffwelder-ellenbytech** [comment]
@jilaypandya Ran a quick test on my hardware with your changes, everything seemed to work as expected. 
My application did a fetch all and got lux, als, and white channels with no issues. :)

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
Sorry this fell through the cracks.

Please rebase and split the unrelated whitespace changes to a separate commit.

File: drivers/sensor/vishay/veml7700/veml7700.c [hunk 1]
```diff
@@ -135,16 +145,24 @@ static void veml7700_sleep_by_integration_time(const struct veml7700_data *data)
 	}
 }
 
-static uint32_t veml7700_counts_to_lux(const struct veml7700_data *data,
-				       uint16_t counts)
+static int veml7700_counts_to_lux(const struct veml7700_data *data, uint16_t counts,
+				  uint32_t *return_value)
```
Comment (MaureenHelm) [inline drivers/sensor/vishay/veml7700/veml7700.c:149]: Rename `return_value` to `lux`

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
Some minor nits, but otherwise looks good

**Review by MaureenHelm — APPROVED** [review]

**Review by teburd — APPROVED** [review]

**Review by dipakgmx — APPROVED** [review]

**Pull request merged**